Refactor: Created a separate widget for Save/Update button and updated add_or_update_alarm_view.dart accordingly

amaydixit11 opened the pull request:
### Description
This PR refactors the `Save or Update button` in `AddOrUpdateAlarmView` widget by creating a separate widget for the button in a different file, while also differentiating the business logic from UI to improve code readability, maintainability, and separation of concerns. 

### Proposed Changes
- Created a separate widget file for the SaveUpdateButton, moving it out of AddOrUpdateAlarmView to enhance modularity.
- Separated UI and business logic by keeping the button's responsibility limited to rendering, while delegating logic to controller functions.
- Extracted permission checks into a helper method for cleaner onPressed logic.
- Refactored alarm creation and update logic into dedicated methods, this eases further refactoring, for example if all the logic is to be moved to a service in the future, we'll just have to move the methods there instead of extracting them from the UI code
- Improved readability by reducing deeply nested conditions and making function calls more descriptive.

## Fixes #654 

## Checklist
- [x] Tests have been added or updated to cover the changes
- [x] Documentation has been updated to reflect the changes
- [x] Code follows the established coding style guidelines
- [x] All tests are passing

MarkisDev commented:
@amaydixit11 Since we're taking up this task to refactor and modularize, let's introduce a new folder for the different components. We should ideally look into how we can make these generic - for ex, right now, the button is still tied to the saveOrUpdate screen. But we use buttons throughout the application. Same goes for a lot of the popups. The idea is to centralize these components in a different folder, so we can keep their behavior, look and feel consistent throughout the application. 

We should be able to use this one button component throughout the application :)

amaydixit11 commented:
@MarkisDev i have implemented a centralized button, it's running great, just wanted to clear a few things 

1. In the app directory,  i've added a new folder "Widgets" for this generic widgets that i'll be creating as a part of refactoring, I believe this is the correct place for such a folder, however if you think it needs to be somewhere else, kindly tell me 
2. Is is better to replace the code with this new generic button everywhere in the code in this single PR? or should i just add the button implementation in this PR and the refactoring in a new one?
